v2/dtos/common/version.go

```diff
@@ -1,21 +1,48 @@
 //
 // Copyright (C) 2020 IOTech Ltd
+// Copyright (C) 2020 Intel Corporation
 //
 // SPDX-License-Identifier: Apache-2.0
 
 package common
 
+import v2 "github.com/edgexfoundry/go-mod-core-contracts/v2"
+
 // VersionResponse defines the latest version supported by the service.
 // This object and its properties correspond to the VersionResponse object in the APIv2 specification:
 // https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-data/2.x#/VersionResponse
 type VersionResponse struct {
-	BaseResponse `json:",inline"`
-	Version      string `json:"version"`
+	Versionable `json:",inline"`
+	Version     string `json:"version"`
 }
 
 // VersionSdkResponse defines the latest sdk version supported by the service.
 // This object and its properties correspond to the VersionSdkResponse object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-data/2.x#/VersionSdkResponse
 type VersionSdkResponse struct {
 	VersionResponse `json:",inline"`
 	SdkVersion      string `json:"sdk_version"`
 }
+
+// NewVersionResponse creates new VersionResponse with all fields set appropriately
+func NewVersionResponse(version string) VersionResponse {
+	return VersionResponse{
+		Versionable: Versionable{
+			ApiVersion: v2.ApiVersion,
+		},
+		Version: version,
+	}
+}
+
+// NewVersionSdkResponse creates new VersionSdkResponse with all fields set appropriately
+func NewVersionSdkResponse(appVersion string, sdkVersion string) VersionSdkResponse {
+	return VersionSdkResponse{
+		VersionResponse: VersionResponse{
+			Versionable: Versionable{
+				ApiVersion: v2.ApiVersion,
+			},
+			Version: appVersion},
+
```

Comment:
could you call  NewVersionResponse() here, such that the block becomes:
```
VersionResponse: NewVersionResponse(appVersion),
```

Reply:
Yes! Excellent idea. I even took it further to Versionable and BaseResponse.